packages/spotlight/src/server/parser/processEnvelope.ts

```diff
@@ -1,11 +1,12 @@
 import type { Envelope, EnvelopeItem } from "@sentry/core";
 import { type UUID, uuidv7obj } from "uuidv7";
 import { RAW_TYPES } from "../constants.ts";
+import { type SourceType, inferEnvelopeSource } from "../formatters/human/utils.ts";
 import { logger } from "../logger.ts";
 import type { RawEventContext } from "./types.ts";
 
 export type ParsedEnvelope = {
-  envelope: [Envelope[0] & { __spotlight_envelope_id: UUID }, Envelope[1]];
+  envelope: [Envelope[0] & { __spotlight_envelope_id: UUID; __spotlight_inferred_source?: SourceType }, Envelope[1]];
```

Comment:
We may wanna create a new mixin type for the additions: `SpotlightEnvelopeExtensions`